### DISCUSSION TITLE
Vercel KV or PostgreSQL as a vector database?

### DISCUSSION THREAD

**Discussion by tomasz-stefaniak** [vercel/community#4946] [Feature Question] [answered]
I'd like to use the KV database for [vector search](https://redis.io/docs/get-started/vector-database/.). Am I right to assume this is currently not supported? Is there any way I could get this feature to work with the KV database, even if it's hacky?

I currently use two databases on my project. One is a MongoDB database hosted on Digital Ocean and the other is Vercel's KV. I'm trying to implement vector search without adding a new database into the mix.

EDIT:Alternatively, can that be done with [Vercel's PostgreSQL](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing) database? It seems to be hosted on Neon which [does support pgvector](https://neon.tech/docs/extensions/pgvector) 🤔

**Comment by leerob** [comment, marked as answer]
You can use Vercel Postgres for this: https://vercel.com/templates/next.js/postgres-pgvector

**Reply by 0xinhua** [reply]
Hi @leerob, I was wondering if I could use Vercel Postgres to store vector data? I currently store on supabase with pgvector and would like to migrate directly to Vercel Postgres. Is there any documentation available?

**Reply by leerob** [reply]
Yes, should work exactly the same - it's just normal postgres https://neon.tech/docs/extensions/pgvector

**Comment by biforbitcoin2050** [comment]
hi, is it possible to make pdfs files into KV Database?
